controllers/installation_controller.go

```diff
@@ -143,13 +156,116 @@ func (r *InstallationReconciler) Reconcile(ctx context.Context, req ctrl.Request
+				// NOTE: Stop installation output cr creation
+				return ctrl.Result{}, nil
+			}
+			// TODO: Separate this into it's own func to test and extract what you
+			// can
+			log.V(Log5Trace).Info("creating installation outputs cr")
+			outputs, err := r.CreateInstallationOutputsCR(ctx, inst, resp)
+			if err != nil {
+				log.V(Log4Debug).Error(err, "error creating installation outputs resource")
+				return ctrl.Result{}, err
+			}
+			// TODO: Wrap in a retry? Try to reduce the errors
+			log.V(Log5Trace).Info("setting owner references on outputs cr")
+			controllerutil.SetOwnerReference(inst, outputs, r.Scheme)
+			err = r.Create(ctx, outputs, &client.CreateOptions{})
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+			installOutputs, err := r.CreateStatusOutputs(ctx, outputs, resp)
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+
+			err = r.Status().Update(ctx, installOutputs)
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+			log.V(Log5Trace).Info("successfully created outputs cr")
+			patchInstall := client.MergeFrom(inst.DeepCopy())
+			inst.SetAnnotations(map[string]string{v1.AnnotationInstallationOutput: "true"})
+			log.V(Log5Trace).Info("patching installation cr")
+			return ctrl.Result{}, r.Patch(ctx, inst, patchInstall)
+		}
+	}
+	patchInstallCR := client.MergeFrom(installCr.DeepCopy())
+	return ctrl.Result{}, r.Patch(ctx, installCr, patchInstallCR)
+}
+func (r *InstallationReconciler) CreateStatusOutputs(ctx context.Context, install *v1.InstallationOutput, in *installationv1.ListInstallationLatestOutputResponse) (*v1.InstallationOutput, error) {
+	install.Status = v1.InstallationOutputStatus{
+		Phase: v1.PhaseSucceeded,
+		Conditions: []metav1.Condition{
+			{
+				Type:               v1.InstallationOutputSucceeded,
+				Status:             metav1.ConditionTrue,
+				Reason:             "InstallationOutputCreatedSuccess",
+				LastTransitionTime: metav1.NewTime(time.Now()),
+				Message:            "outputs custom resource generated succeeded",
+			},
+		},
+	}
+
+	outputs := []v1.Output{}
+	outputNames := []string{}
+	for _, output := range in.Outputs {
+		tmpOutput := v1.Output{
+			Name:      output.Name,
+			Type:      output.Type,
+			Sensitive: output.Sensitive,
+			Value:     output.GetValue().GetStringValue(),
+		}
+		outputNames = append(outputNames, output.Name)
+		outputs = append(outputs, tmpOutput)
+	}
+	install.Status.Outputs = outputs
+	install.Status.OutputNames = strings.Join(outputNames, ",")
+
+	return install, nil
+}
+
+func (r *InstallationReconciler) CreateInstallationOutputsCR(ctx context.Context, install *v1.Installation, in *installationv1.ListInstallationLatestOutputResponse) (*v1.InstallationOutput, error) {
+	if len(in.Outputs) < 1 {
+		return nil, fmt.Errorf("no outputs for the installation %s", install.Name)
```

Comment:
Instead of error, maybe best to log?
Some installations may not have outputs

Reply:
Could the installationoutput cr have an owner ref to the installation?

Reply:
> Instead of error, maybe best to log?

This only makes the CR if the outputs exist from the previous call.  

> Some installations may not have outputs

It errors only to requeue in the event that the output length being returned from the grpc server is less than 1 .  I agree, I should probably still log.  I have to short circuit in the event that the output doesn't exist or it'll make a CR with a blank output.